packages/next/export/index.ts

```diff
@@ -672,14 +672,18 @@ export default async function exportApp(
           )
           const jsonDest = join(pagesDataDir, `${route}.json`)
 
-          await promises.mkdir(dirname(htmlDest), { recursive: true })
-          await promises.mkdir(dirname(jsonDest), { recursive: true })
+          await Promise.all([
+            promises.mkdir(dirname(htmlDest), { recursive: true }),
+            promises.mkdir(dirname(jsonDest), { recursive: true })
+          ])
 
           const htmlSrc = `${orig}.html`
           const jsonSrc = `${orig}.json`
 
-          await promises.copyFile(htmlSrc, htmlDest)
-          await promises.copyFile(jsonSrc, jsonDest)
+          await Promise.all([
+            promises.copyFile(htmlSrc, htmlDest),
+            promises.copyFile(jsonSrc, jsonDest)
+          ])
```

Comment:
I wonder if there could be even more optimization: instead of creating all dirs and then copying all files, we could split the tasks, like:

```js
const htmlSrc = `${orig}.html`
const jsonSrc = `${orig}.json`

await Promise.all([
  promises.mkdir(dirname(htmlDest), { recursive: true }).then(() => promises.copyFile(htmlSrc, htmlDest)),
  promises.mkdir(dirname(jsonDest), { recursive: true }).then(() => promises.copyFile(jsonSrc, jsonDest))
])
```